### ISSUE TITLE
General type definition for kintone

### ISSUE THREAD

**Issue by the-red** [kintone/dts-gen#116]
I'd like to define type by each variable as follows.
```ts
  type RecordNumber = {
    type: 'RECORD_NUMBER'
    value: string
    error?: string
  }
  const id: RecordNumber = event.record.レコード番号
```

It is useful if you provide a general type definition for kintone 😄 
```ts
  const id: kintone.types.RecordNumber = event.record.レコード番号
```
How about preparing the following d.ts files as standard?
```ts
declare namespace kintone.types {
  interface SimpleValueTypes {
    value: string
    disabled?: boolean
    error?: string
  }
  interface SingleLineText extends SimpleValueTypes {
    type: 'SINGLE_LINE_TEXT'
  }
  interface MultiLineText extends SimpleValueTypes {
    type: 'MULTI_LINE_TEXT'
  }
  interface RichText extends SimpleValueTypes {
    type: 'RICH_TEXT'
  }
  interface Date extends SimpleValueTypes {
    type: 'DATE'
  }
  interface Number extends SimpleValueTypes {
    type: 'NUMBER'
  }
  interface Datetime extends SimpleValueTypes {
    type: 'DATETIME'
  }
  interface Time extends SimpleValueTypes {
    type: 'TIME'
  }
  interface RecordNumber extends SimpleValueTypes {
    type: 'RECORD_NUMBER'
  }
  interface DropDown extends SimpleValueTypes {
    type: 'DROP_DOWN'
  }
  interface Link extends SimpleValueTypes {
    type: 'LINK'
  }
  interface Calc extends SimpleValueTypes {
    type: 'CALC'
  }
  interface RadioButton extends SimpleValueTypes {
    type: 'RADIO_BUTTON'
  }

  interface StringListTypes {
    value: string[]
    disabled?: boolean
    error?: string
  }
  interface CheckBox extends StringListTypes {
    type: 'CHECK_BOX'
  }
  interface MultiSelect extends StringListTypes {
    type: 'MULTI_SELECT'
  }

  interface EntityListType {
    value: { code: string; name: string }[]
    disabled?: boolean
    error?: string
  }
  interface UserSelect extends EntityListType {
    type: 'USER_SELECT'
  }
  interface GroupSelect extends EntityListType {
    type: 'GROUP_SELECT'
  }
  interface OrganizationSelect extends EntityListType {
    type: 'ORGANIZATION_SELECT'
  }

  interface File {
    type: 'FILE'
    value: {
      contentType: string
      fileKey: string
      name: string
      size: string
    }[]
    disabled?: boolean
    error?: string
  }
}
```

**Comment by the-red** [comment]
I have referenced this code
https://github.com/kintone/dts-gen/blob/bb07721affb8c66ac2044957c990031aa949b697/src/converters/fileldtype-converter.ts#L12-L44

and this pull request.
https://github.com/kintone/dts-gen/pull/111/files

**Comment by koba04** [comment]
@the-red 
Thank you!
After #111 has been merged, we'll consider to export types you mention as General type.

**Comment by zaki-yama** [comment]
@the-red 
Thank you for your feedback!
We released version `2.2.0` which provides general type definitions as `kintone.fieldTypes.XXX`.
